lib/bundler/resolver.rb

```diff
@@ -131,7 +131,7 @@ def source
       def for?(platform, required_ruby_version)
         if spec = @specs[platform]
           if required_ruby_version && spec.respond_to?(:required_ruby_version) && spec_required_ruby_version = spec.required_ruby_version
-            spec_required_ruby_version.satisfied_by?(required_ruby_version.gem_version)
+            spec_required_ruby_version.satisfied_by?(Gem::Version.new("#{required_ruby_version.gem_version}.#{required_ruby_version.patchlevel}"))
```

Comment:
`required_ruby_version.patchlevel` can be nil, and this should be a method on `Bundler::RubyVersion`